model/task_lifecycle.go

```diff
@@ -733,6 +685,36 @@ func UpdateBuildAndVersionStatusForTask(taskId string, updates *StatusChanges) e
 		"num_tasks":     len(buildTasks),
 	})
 
+	startPhaseAt = time.Now()
+	for _, displayTask := range cache.List() {
+		if err = UpdateDisplayTask(displayTask); err != nil {
```

Comment:
It might be helpful to instrument `UpdateDisplayTask` and `UpdateCachedTask`, since these are database operations in a loop, and therefore also might be slow.

Reply:
The individual queries aren't too interesting - the performance problems are due to the aggregate of them